File: Custom_responses_that_don’t_hit_origin_servers/lua/deny_a_spider.lua
local _M = {}

local ROBOT_REG = 'robot'

function _M.handler(event)
    local request = event.request
    local headers = request.headers

    -- 如果 UA 中含有 robot 关键字, 返回 403
    if type(headers['user-agent']) == 'string' then
        if re.find(headers['user-agent'], ROBOT_REG, 'jio') then
            return {status = 403}
        end
    end

    -- 其他继续回源
    return event.request
end

return _M

File: Custom_responses_that_don’t_hit_origin_servers/python/deny_a_spider.py
import re

ROBOT_REG = '.*robot.*'


async def handler(event):
    request = event.request
    headers = request.headers

    # 如果 UA 中含有 robot 关键字, 返回 403
    if 'user-agent' in headers:
        ua = headers.get('user-agent')
        if re.match(ROBOT_REG, ua, re.I):
            return {'status': 403}

    # 其他继续回源
    return Request(request=request)
